### PR TITLE
Updated TKR version to match what is in the next TKG Bom

### PR DESCRIPTION
**Pull request by saji-pivotal**
### What this PR does / why we need it
The CI pipelines have been failing ever since we started using the 1.7.0 TKG BOM. This is likely because the 1.7.0 TKG BOM is using `v1.23.8.vmware.2-tkg.2-zshippable` and not `v1.23.8.vmware.2-tkg.1-zshippable`. 

This was causing the integration tests to fail. This PR changes the TKR version to `v1.23.8.vmware.2-tkg.2-zshippable` in order to get the CI tests passing

This PR also adds a helper function to the integration tests which waits for the number of scaled up nodes to be greater than / equal to a target number of nodes, as opposed to being exactly equal to the target number of nodes. This was done because the autoscaler tests were failing when the number of cluster nodes was 4, but the expected number was 3

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
1. Relying on the CI checks to pass before merging this PR

### Release note

```release-note

```

### CODE REVIEW

**Review by vuil — APPROVED** [review]
lgtm, thanks

**Comment by vuil** [comment]
lgtm. @saji-pivotal please update the title or description so it is clear what tests were broken that this is attempting to fix.

**Review by tenczar — COMMENTED** [review]
small change

File: pkg/v1/providers/infrastructure-docker/v1.1.5/yttcc/overlay.yaml [hunk 1]
```diff
@@ -20,7 +20,7 @@ metadata:
   name: #@ data.values.CLUSTER_NAME
   labels:
     tkg.tanzu.vmware.com/cluster-name: #@ data.values.CLUSTER_NAME
-    run.tanzu.vmware.com/tkr: v1.23.8---vmware.2-tkg.1-zshippable
+    run.tanzu.vmware.com/tkr: v1.23.8---vmware.2-tkg.2-zshippable
 spec:
   topology:
     class: #@ data.values.CLUSTER_CLASS
```
[the remark this answers is not in the document]
Comment (vuil) [reply pkg/v1/providers/infrastructure-docker/v1.1.5/yttcc/overlay.yaml:26]: nice catch!

**Review by imikushin — APPROVED** [review]
LGTM

**Comment by saji-pivotal** [comment]
This needed to be updated as well - https://github.com/vmware-tanzu/tanzu-framework/blob/main/tkg/test/tkgctl/shared/constants.go#L31

**Comment by vuil** [comment]
@saji-pivotal you wanna rebase now that https://github.com/vmware-tanzu/tanzu-framework/pull/3552 is merged?

**Review by anujc25 — APPROVED** [review]

**Pull request merged**